src/test/java/org/elasticsearch/cluster/MinimumMasterNodesTests.java

```diff
@@ -126,29 +122,26 @@ public boolean apply(Object obj) {
         assertThat(state.nodes().size(), equalTo(2));
         assertThat(state.metaData().indices().containsKey("test"), equalTo(true));
 
-        logger.info("Running Cluster Health");
-        ClusterHealthResponse clusterHealth = client().admin().cluster().health(clusterHealthRequest().waitForGreenStatus()).actionGet();
-        logger.info("Done Cluster Health, status " + clusterHealth.getStatus());
-        assertThat(clusterHealth.isTimedOut(), equalTo(false));
-        assertThat(clusterHealth.getStatus(), equalTo(ClusterHealthStatus.GREEN));
+        ensureGreen();
 
-        logger.info("--> verify we the data back");
+        logger.info("--> verify we the data back after cluster reform");
         for (int i = 0; i < 10; i++) {
-            assertThat(client().prepareCount().setQuery(QueryBuilders.matchAllQuery()).execute().actionGet().getCount(), equalTo(100l));
+            assertHitCount(client().prepareCount().setQuery(QueryBuilders.matchAllQuery()).execute().actionGet(), 100);
         }
 
         internalCluster().stopRandomNonMasterNode();
-        assertThat(awaitBusy(new Predicate<Object>() {
+        assertBusy(new Runnable() {
             @Override
-            public boolean apply(Object obj) {
+            public void run() {
                 ClusterState state = client().admin().cluster().prepareState().setLocal(true).execute().actionGet().getState();
-                return state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID);
+                assertThat(state.blocks().hasGlobalBlock(DiscoverySettings.NO_MASTER_BLOCK_ID), equalTo(true));
             }
-        }), equalTo(true));
+        });
 
         logger.info("--> starting the previous master node again...");
         internalCluster().startNode(settings);
 
+        ensureGreen();
```

Comment:
I added it to get a better message if we fail to wait for green, we anyhow wait for green in the next line like you observed, but we miss on better logging if we don't reach green. I want to improve this options and allow to specify number of nodes to the test framework, but prefer to do it in a different change